### DISCUSSION TITLE
Eleventy and Web Components

### DISCUSSION THREAD

**Discussion by bsastregx** [11ty/eleventy#2510] [Q&A]
What is the best approach/library for using web components with eleventy?
I mean web components with encapsulated styles and logic, not just nunjucks templates, as I seen in some tutorials.
Have to say that my preference is StencilJS, but I am having a difficult time integrating it to eleventy.

**Comment by ruiconti** [comment]
I've had a somewhat good experience with Lit. They are also doing some recent investments on SSR that has some promising, frictionless integration with 11ty: https://lit.dev/blog/2022-02-07-eleventy/.

**Reply by bsastregx** [reply]
@ruiconti thanks!
